### DISCUSSION TITLE
[Bug] GRPC serverSide java.lang.IllegalArgumentException: ContextSnapshot can't be null.

### DISCUSSION THREAD

**Discussion by MrCoderYu** [apache/skywalking#12606] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

grpc serverSide occurs 'java.lang.IllegalArgumentException: ContextSnapshot can't be null.' when using ThreadPoolTaskExecutor

### What you expected to happen

Compatible with GRPC interface using thread pool mode

### How to reproduce

    @Override
    public void grpcMethod(Request request, StreamObserver<Response> responseObserver) {
        CompletableFuture.runAsync(() -> {
            // ...bizCode
        }, userRequestHandleTaskExecutor);
    }

### Anything else

V  9.3.0
TracingServerCall:ContextManager.continued(ServerInterceptor.CONTEXT_SNAPSHOT_KEY.get());

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
You could check test scenarios in the agent projects. 
If this case is not verified, then we can't guarantee it works. We may need some relative patches for your scenarios.

**Reply by wu-sheng** [reply]
As I am not a user for this scenario, either you need to learn source codes and debug, or you have to wait for others to step in.

**Comment by hzihui** [comment]
I have also encountered this problem. The 'close' method of 'TraceServerCall' is null when obtaining 'ContextSnapshot' data through 'Context. Key<ContextSnapshot>'. The reason is that 'Grpc Context' uses ThreadLocal. Can you provide a temporary solution?

**Reply by wu-sheng** [reply]
Until there is a new plugin, you have to remove the grpc plugin, and accept the tracing will be broken by this.

**Reply by hzihui** [reply]
Is there currently a plan to fix this issue?

**Reply by wu-sheng** [reply]
SkyWalking community relies on everyone as volunteers to fix this. There is no plan if you mean me.

**Reply by hzihui** [reply]
Okay, thank you. I'll try to solve this problem
